pandas/core/index.py

```diff
@@ -1412,12 +1412,7 @@ def shift(self, periods=1, freq=None):
         -------
         shifted : Index
         """
-        if periods == 0:
-            # OK because immutable
-            return self
-
-        offset = periods * freq
-        return Index([idx + offset for idx in self], name=self.name)
+        raise NotImplementedError
```

Comment:
raise with a helpful message (e.g. say something about not supported for this type (e.g. show the type))